app/controllers/characters_controller.rb

```diff
@@ -69,6 +69,19 @@ def update
     end
   end
 
+  def duplicate
+    @dup = @character.dup
```

Comment:
Standard is using save! instead of save or create!/create or similar equivalents